bridges/otellogrus/hook_test.go

```diff
@@ -0,0 +1,228 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+package otellogrus
+
+import (
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+
+	"go.opentelemetry.io/otel/log"
+	"go.opentelemetry.io/otel/log/logtest"
+)
+
+func TestNewHook(t *testing.T) {
+	assert.NotNil(t, NewHook())
+}
+
+func TestHookLevels(t *testing.T) {
+	for _, tt := range []struct {
+		name    string
+		options []Option
+
+		wantLevels []logrus.Level
+	}{
+		{
+			name:       "with the default levels",
+			wantLevels: logrus.AllLevels,
+		},
+		{
+			name: "with provided levels",
+			options: []Option{
+				WithLevels([]logrus.Level{logrus.PanicLevel}),
+			},
+			wantLevels: []logrus.Level{logrus.PanicLevel},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			levels := NewHook(tt.options...).Levels()
+			assert.Equal(t, tt.wantLevels, levels)
+		})
+	}
+}
+
+func TestHookFire(t *testing.T) {
+	now := time.Now()
+
+	for _, tt := range []struct {
+		name  string
+		entry *logrus.Entry
+
+		wantRecords map[string][]log.Record
+		wantErr     error
+	}{
+		{
+			name:  "emits an empty log entry",
+			entry: &logrus.Entry{},
+
+			wantRecords: map[string][]log.Record{
+				bridgeName: {
+					buildRecord(log.StringValue(""), time.Time{}, 0, nil),
+				},
+			},
+		},
+		{
+			name: "emits a log entry with a timestamp",
+			entry: &logrus.Entry{
+				Time: now,
+			},
+			wantRecords: map[string][]log.Record{
+				bridgeName: {
+					buildRecord(log.StringValue(""), now, 0, nil),
+				},
+			},
+		},
+		{
+			name: "emits a log entry with severity level",
+			entry: &logrus.Entry{
+				Level: logrus.FatalLevel,
+			},
+			wantRecords: map[string][]log.Record{
+				bridgeName: {
+					buildRecord(log.StringValue(""), time.Time{}, log.SeverityTrace1, nil),
+				},
+			},
+		},
+		{
+			name: "emits a log entry with data",
+			entry: &logrus.Entry{
+				Data: logrus.Fields{
+					"hello": "world",
+				},
+			},
+			wantRecords: map[string][]log.Record{
+				bridgeName: {
+					buildRecord(log.StringValue(""), time.Time{}, 0, []log.KeyValue{
+						log.String("hello", "world"),
+					}),
+				},
+			},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			rec := logtest.NewRecorder()
+
+			err := NewHook(WithLoggerProvider(rec)).Fire(tt.entry)
+			assert.Equal(t, tt.wantErr, err)
+
+			for k, v := range tt.wantRecords {
+				found := false
+
+				for _, s := range rec.Result() {
+					if k == s.Name {
+						assert.Equal(t, v, s.Records)
```

Comment:
`log.Record` is not comparable. It would be safer to not use `testify` which uses reflection to check for eqaulity and create a helper function instead.